libnetwork/endpoint.go

```diff
@@ -575,7 +575,7 @@ func (ep *Endpoint) sbJoin(ctx context.Context, sb *Sandbox, options ...Endpoint
 	}()
 
 	if err := sb.populateNetworkResources(ctx, ep); err != nil {
-		return err
+		return errdefs.System(err)
```

Comment:
We should probably make all functions be a proper errdef (but check if any of the errors would already be typed), or if none of them are typed, we could consider the function as a whole to always be a system error (assuming there's no user-input involved that would require it to be a user-error ("invalid parameter")).

It looks like it's called in 2 places; one of those shadows the underlying error (so this `errdefs.System` would go lost, because it's using `%v`);
https://github.com/moby/moby/blob/d313bb58c433cfb9ff91d4c376c53c0fa54a058a/libnetwork/default_gateway.go#L88-L90

```go
	if err = newEp.sbJoin(context.TODO(), sb); err != nil {
		return errdefs.System(fmt.Errorf("container %s: endpoint join on GW Network failed: %v", sb.containerID, err))
	}
```


The other use is in the exported `endpoint.Join`, which returns it as-is, so could be used (again, assuming any error here should be considered an internal error) to type it; https://github.com/moby/moby/blob/d313bb58c433cfb9ff91d4c376c53c0fa54a058a/libnetwork/endpoint.go#L488


```go
if err := ep.sbJoin(ctx, sb, options...); err != nil {
    return errdefs.System(err)
}
return nil
```